cli/src/java/org/apache/hadoop/hive/cli/CliDriver.java

```diff
@@ -499,7 +499,10 @@ public CommandProcessorResponse processReader(BufferedReader r) throws IOExcepti
     while ((line = r.readLine()) != null) {
       // Skipping through comments
       if (! line.startsWith("--")) {
-        qsb.append(line + "\n");
+        if (line.contains("--") && line.endsWith(";")) {
```

Comment:
I'm wondering what will happen with:
```
select 'hello -- there' ;
```
I think we only use the `cli` stuff during testing - the `hive` command is launching a `beeline` shell;
can a user still launch the `hive` cli?